### ISSUE TITLE
URL doesn't exist / OC7.0.2 - MyOwnNotes 1.5 - Notes 0.9

### ISSUE THREAD

**Issue by Trollken** [aykit/MyOwnNotes#83]
Hi guys,

I got a similar problem to #74.
I just installed MyOwnNotes 1.5 (via f-droid), and Notes 0.9 on my Owncloud (recently updated to 7.0.2).

Self signed certificate added on my Android smartphone too (other apps like DAVdroid and Owncloud client work well).

But when i try to sync MyOwnNotes with owncloud : « L'adresse internet n'existe pas. Vérifiez la connection internet et l'orthographe » (fr).

Dunno what to do :/.

(also, i changed my https port, it's not 443 !
My owncloud address is a bit weird, i probably should try to change it for something more "simple".
Atm, it's like : https://my.ip.add.ress:port/owncloud
But, it's working with every other app i use ! )

**Comment by oysteine** [comment]
Maltefiala: depends on what you mean by malformed? I've used 192.168.1.2 as the common name for the certificate, as I do not have DNS set up.This is not malformed according to RFC 5280, but there might be limitations in myownnotes' implementation?

**Comment by praet0ri4n** [comment]
Ehm, sorry for not beeing clear... I meant the Common Name in my previous self signed certificate was "issued" to another host or IP. It was the original snakeoil cert that comes with apache ssl installation in ubuntu. So I created a new pair of server key and cert and entered my own domain name, e.g. "example.com" for common name. Yes, IP can be added too, but I guess not the local LAN ip 192.168... but your public IP. If you've no static public IP it could be changing and you would need to create new ssl cert everytime. Therefore it's better to use a domain name. So much for SSL cert and Common Name....

I was looking at the logcat on my android phone and there was an error message stating that the ssl certificate has some errors and no connection can be established to my server... even after adding the certificate to my keychain (see the procedure on the apps website). There comes the "URL doesn't exist.." message from!

After I added the new cert to my phone, connection was established and notes synced.

**Comment by oysteine** [comment]
I only access my owncloud instance on my own network, or through a VPN to my own network so 192.168... is just as it should be. Then it seems your fix is not relevant to my problem, but nice to know.

**Comment by Trollken** [comment]
Hi guys,

Well, yesterday i updated my owncloud server to 7.0.4, and had to create new cert (previous one expired) and everything is working fine now !

Thks

**Comment by oysteine** [comment]
Upgraded to owncloud 8, no change for MyOwnNotes on Android, still getting the  "The URL you request data from doesn't seem to exist. Check spelling or your internet connection".  message. Will try again on the inside of the VPN.

**Comment by ramschmaerchen** [comment]
@Trollken : Great

@oysteine : I am pretty sure this has to do with a cert error or your vpn setup. I don't think that we can help much here, sorry

**Comment by predus2** [comment]
Hi,
I install oc 8 + MyOwnNotes1.5 + ocnotes 1. I use owncloud to syc contact, callendar + sms, acces via web, but can't acces notes. Still Check spelling or your internet connection message. 
settings:
https://193.xxx.xxx.xxxx:1857/owncloud
maybe port forwarding problem?

can you help me?

**Comment by flipsa** [comment]
I'm seeing the same problem and have a similar setup to @oysteine (owncloud-8.0.2 on my LAN; either connect from inside my LAN to https://192.168.1.254/owncloud or use openvpn to connect to my home LAN first). I did of course successfully import my certificate (CN is 192.168.1.xxx) into Android, and the cert works fine with other apps like e.g. DAVDroid. 

The toast message says i mis-spelled the URL and LogCat throws a "Hostname 192.168.1.254 was not verified". An IOException is thrown." at me. 
 
I can provide more info, but since my setup and error are the same other people see (with exactly the same debug messages and all), I guess that won't help much at this point. Handling of self-signed certs in combination with private IPv4 address spaces (RFC1918) seems broken. Too bad I can't decide to use http instead (which is not a security problem, as I'm connecting through my VPN when not at home).

**Comment by maltefiala** [comment]
Please look at this issue, maybe it helps: #112

**Comment by mke21** [comment]
I also got this problem. I've got an owncloud server (version 8.2) running on my QNAP nas. I inserted a self-signed certificate which I imported in my Android telephone. The owncloud client works fine with this certificate and als syncing contacts en agenda seems fine. But MyOwnNote (versio 1.6) tells me the url doesn't seem to exist. Which is of course untrue. I am pretty sure that the Common Name in the certificate is correct.
